### PR TITLE
gh-125674: Doc: Fix type of `newfunc` first parameter

### PR DESCRIPTION
**Pull request by rhansen**
* Issue: gh-125674

----
📚 Documentation preview 📚: https://cpython-previews--125675.org.readthedocs.build/

### CODE REVIEW

File: Misc/NEWS.d/next/Documentation/2024-10-17-22-26-14.gh-issue-125674.uM2SgY.rst
Comment (AA-Turner) [inline Misc/NEWS.d/next/Documentation/2024-10-17-22-26-14.gh-issue-125674.uM2SgY.rst:1]: You don't need NEWS entries for docs PRs, I've added skip-news.

**Review by AA-Turner — APPROVED** [review]

File: Misc/NEWS.d/next/Documentation/2024-10-17-22-26-14.gh-issue-125674.uM2SgY.rst
Comment (rhansen) [reply Misc/NEWS.d/next/Documentation/2024-10-17-22-26-14.gh-issue-125674.uM2SgY.rst:1]: Good to know, thank you!

File: Doc/c-api/typeobj.rst [hunk 1]
```diff
@@ -2645,7 +2645,7 @@ Slot Type typedefs
 
    See :c:member:`~PyTypeObject.tp_free`.
 
-.. c:type:: PyObject *(*newfunc)(PyObject *, PyObject *, PyObject *)
+.. c:type:: PyObject *(*newfunc)(PyTypeObject *, PyObject *, PyObject *)
```
Comment (rruuaanng) [inline Doc/c-api/typeobj.rst:2650]: I'm sorry, but let me ask a question. Why can't they just be `PyObject`, but instead unify the code and documentation to `PyTypeObject`?

Comment (rhansen) [reply Doc/c-api/typeobj.rst:2650]: @rruuaanng I'm not sure what you mean.  This PR *is* unifying the documentation to match the code (`PyTypeObject`).

**Review by willingc — APPROVED** [review]

**Pull request merged**